src/components/AlertsMenu/AlertsMenu.tsx

```diff
@@ -0,0 +1,126 @@
+'use client';
+
+import { Button } from '@nextui-org/button';
+import { Image } from '@nextui-org/image';
+import { Popover, PopoverContent, PopoverTrigger } from '@nextui-org/popover';
+import { useMemo } from 'react';
+
+import { useSidebar } from '../Sidebar/SidebarContext';
+
+export type AlertKey = 'hunger' | 'conflicts1' | 'conflicts2' | 'hazards';
+
+export type AlertTypeKey = 'conflicts';
+
+type Alert = {
+  key: AlertKey;
+  label: string;
+  icon: string;
+};
+
+type AlertType = {
+  key: AlertTypeKey;
+  label: string;
+  icon: string;
+  subalerts: Alert[];
+};
+
+const alertTypes: (Alert | AlertType)[] = [
+  {
+    key: 'hunger',
+    label: 'Hunger Alert',
+    icon: '/menu_fcs.png',
+  },
+  {
+    key: 'conflicts',
+    label: 'Conflicts',
+    icon: '/menu_conflicts.png',
+    subalerts: [
+      {
+        key: 'conflicts1',
+        label: 'Conflicts 1',
+        icon: '/menu_conflicts.png',
+      },
+      {
+        key: 'conflicts2',
+        label: 'Conflicts 2',
+        icon: '/menu_conflicts.png',
+      },
+    ],
+  },
+  {
+    key: 'hazards',
+    label: 'Hazards',
+    icon: '/menu_hazards.png',
+  },
+];
+
+type AlertsMenuProps = {
+  variant: 'outside' | 'inside';
+};
+
+export default function AlertsMenu({ variant }: AlertsMenuProps) {
+  const { isAlertSelected, toggleAlert } = useSidebar();
+
+  const isSubAlertClicked = useMemo(
+    () => (mainAlert: AlertTypeKey) => {
+      const alert = alertTypes.find((a) => a.key === mainAlert) as AlertType;
+      return alert.subalerts.some((subalert) => isAlertSelected(subalert.key)) ?? false;
+    },
+    [isAlertSelected]
+  );
+
+  return (
+    <div className="flex gap-1">
+      {alertTypes.map((item) =>
+        'subalerts' in item ? (
+          <Popover placement={variant === 'inside' ? 'bottom' : 'top'} key={item.key}>
+            <PopoverTrigger>
+              <Button isIconOnly radius="full" className={isSubAlertClicked(item.key) ? 'bg-primary' : 'bg-background'}>
+                <div className="w-6 h-6 flex items-center justify-center">
+                  <Image
+                    src={item.icon}
+                    alt={item.label}
+                    className="object-contain w-auto h-auto max-w-full max-h-full"
+                  />
+                </div>
+              </Button>
```

Comment:
it seems like this is repeated 3 times in this file. Can't we create a component from this? Olcick and className could be set with props

Reply:
I've tried but couldn't make the popover trigger work when I provide it with my custom button function component

Reply:
You should provide a function in the custom component props

Reply:
Solved: had to forward the ref